ahk/autohotkey.py

```diff
@@ -1,18 +1,31 @@
+import atexit
 from collections import deque
 from ahk.mouse import MouseMixin
 from ahk.window import Window, WindowMixin
 from ahk.script import ScriptEngine
 from ahk.screen import ScreenMixin
 from ahk.keyboard import KeyboardMixin
 from ahk.sound import SoundMixin
+from ahk.communication import EventListener
 
 class AHK(WindowMixin, MouseMixin, KeyboardMixin, ScreenMixin, SoundMixin):
-    pass
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(self, *args, **kwargs)
+        is_main = kwargs.pop("use_event_loop", None)
+        if is_main == None or is_main == True:
+            self.EventListener = EventListener()
+        atexit.register(self._on_exit)
+        
+    def _on_exit(self):
+        if self.EventListener:
+            self.EventListener.stop()
 
 
 class ActionChain(AHK):
     def __init__(self, *args, **kwargs):
         self._actions = deque()
+        kwargs.append("use_event_loop", True)
```

Comment:
this might seem strange, but it is an explicit true to if it should make it's own event loop, this might not be required however. It all depends on how ActionChain develops, and it is likely that hotkeys aren't part of the intended use of ActionChain.